### PR TITLE
Feature/api test create user

### LINKED ISSUE #4: API Testing
**Is your feature request related to a problem? Please describe.**
As a developer of this project, I want to know if the code I wrote broke anything anywhere in the API.

**Describe the solution you'd like**
Using Jest, to have automated testing on the API setup in Travis CI. It should test all the queries and mutations in the API.

### PR DESCRIPTION
**Pull request by tristanMatthias**
- Fixes #4 
- Added testing structure to API
- Refactored database module so it can be used in testing
- Added basic tests for `createUser` mutation

### CODE REVIEW

**Comment by tristanMatthias** [comment]
@NoSoEpicCoder , add a comment here so I can request your review too :)

File: packages/api/src/Database/Database.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+   * Loops over all tables and removes everything!
+   * BE CAREFUL
+   * Used in testing
+   */
+  async DANGEROUSLY_RESET_DATABASE() {
+    Promise.all(this.connection.entityMetadatas.map(async entity => {
+      const repository = await this.getRepository(entity.name);
+      await repository.clear();
+    }));
+  }
```
Comment (suzynakayama) [inline packages/api/src/Database/Database.service.ts:30]: As I mentioned yesterday, I know nothing about testing, trying to learn now, but I have one question: Why would we want to reset the db in a test?

Comment (tristanMatthias) [reply packages/api/src/Database/Database.service.ts:30]: @suzynakayama, I mentioned this on our talk last night. Was this still unclear? The intention behind resetting this is to make sure it's a "clean" set of data each time you test, so it doesn't introduce strange bugs each time. This should also be improved with #53

Comment (suzynakayama) [reply packages/api/src/Database/Database.service.ts:30]: @tristanMatthias , no I wrote this before the talk last night. thank you.

Comment (tristanMatthias) [reply packages/api/src/Database/Database.service.ts:30]: OK great. Can you approve the PR then? I'll merge it in

**Pull request merged**